### PR TITLE
[bsp][airm2m/air105][rsoc] Fix compilation issues with bsp of airm2m/air105

### PR DESCRIPTION
**Pull request by hydevcode**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)
发现airm2m/air105在编译最后一步会出错

#### 你的解决方案是什么 (what is your solution)
修改makesoc.py

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP:bsp/airm2m/air105

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by mysterywolf** [comment]
@Dozingfiretruck 帮忙review一下这个PR呗

**Review by supperthomas — APPROVED** [review]

File: bsp/airm2m/air105/board/board.c [hunk 1]
```diff
@@ -53,10 +54,11 @@ void rt_hw_board_init(void)
 #ifdef RT_USING_HEAP
     rt_system_heap_init((void *)HEAP_BEGIN, (void *)HEAP_END);
 #endif
+    extern void rt_hw_systick_init(void);
```
Comment (mysterywolf) [inline bsp/airm2m/air105/board/board.c:57]: 这个函数的头文件在哪里呀 可否直接引用头文件呀

File: bsp/airm2m/air105/makesoc.py [hunk 2]
```diff
@@ -17,13 +16,13 @@
...
     if not os.path.exists(pack_path+'/soc_download.exe'):
-        urllib.urlretrieve("http://cdndownload.openluat.com/rt-thread/airm2m/air105/soc_download.exe", pack_path+'/soc_download.exe')
+        urllib.request.urlretrieve("http://cdndownload.openluat.com/rt-thread/airm2m/air105/bootloader.bin", pack_path + '/bootloader.bin')
 
     shutil.copy(out_path+bin_file_name, pack_path+'/'+bin_file_name)
     shutil.make_archive(out_file, 'zip', root_dir=pack_path)
     os.remove(pack_path+'/'+bin_file_name)
     os.rename(out_file+'.zip',out_file+'.soc')
-    print('end')
+    print('end')
```
Comment (mysterywolf) [inline bsp/airm2m/air105/makesoc.py:28]: 多敲一个换行

File: bsp/airm2m/air105/libraries/HAL_Driver/Src/core_dma.c [hunk 3]
```diff
@@ -352,7 +352,7 @@ static void DMA_IrqHandle(int32_t IrqLine, void *pData)
             if (DMA->StatusErr_L & (1 << i))
             {
                 DMA->ClearErr_L = (1 << i);
-                hwDMAChannal[i].CB(hwDMAChannal[i].pData, 0xffffffff);
+                hwDMAChannal[i].CB(hwDMAChannal[i].pData,(void *)0xffffffff);
```
Comment (mysterywolf) [inline bsp/airm2m/air105/libraries/HAL_Driver/Src/core_dma.c:355]:
```suggestion
                hwDMAChannal[i].CB(hwDMAChannal[i].pData, (void *)0xffffffff);
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/airm2m/air105/board/board.c (hunk 1 above)
Comment (hydevcode) [reply bsp/airm2m/air105/board/board.c]: 已手动声明函数在对应的头文件，修改好了

File: bsp/airm2m/air105/makesoc.py (hunk 2 above)
Comment (hydevcode) [reply bsp/airm2m/air105/makesoc.py]: 已修改

File: bsp/airm2m/air105/libraries/HAL_Driver/Src/core_dma.c (hunk 3 above)
Comment (hydevcode) [reply bsp/airm2m/air105/libraries/HAL_Driver/Src/core_dma.c]: 已合并

**Comment by Dozingfiretruck** [comment]
因为会网络下载bl所以ci 是否添加评估一下
另外 合宙已经放弃mcu业务，此bsp是否删除也请评估下 
@mysterywolf

**Review by mysterywolf — APPROVED** [review]

**Comment by mysterywolf** [comment]
先保留吧

**Pull request merged**